vernac/loadpath.ml

```diff
@@ -272,6 +272,27 @@ let locate_qualified_library ?root qid :
       Ok (library, file)
     | Error _ as e -> e
 
+let warn_deprecated_missing_stdlib =
+  CWarnings.create ~name:"deprecated-missing-stdlib"
+    ~category:Deprecation.Version.v8_21
+    (fun qid ->
+      Pp.(str "Loading Stdlib without prefix is deprecated." ++ spc ()
+          ++ str "Use \"From Stdlib Require " ++ Libnames.pr_qualid qid
+          ++ str "\"" ++ spc() ++ str "or the deprecated \"From Coq Require "
+          ++ Libnames.pr_qualid qid ++ str "\"" ++ spc ()
+          ++ str "for compatibility with older Coq versions."))
+
+(* temporary handling deprecated loading of stdlib without root *)
```

Comment:
I am not sure I see the point of this hack; if we want `-R` semantics, shouldn't the `coq-stdlib` makefile specify so?

IMHO it is confusing that `coq-stdlib` uses `-Q` but Coq uses `-R`, this means we have to hardcode this on all tooling, including coqdep etc... (which I'm surprised it works without this, someone should check warnings in CI to see if indeed coqdep can locate stuff).

So I'd say if we are doing the splitting, let the `coq-stdlib` package set their `-R/-Q` policy and not `coq-core`.

Reply:
Indeed `coqdep` is broken, see updated comment.

Reply:
This is just a temporary piece of code. The final result after cleanup will be the one you expect.

Reply:
I am not sure what I would expect here. Actually modulo the coqdep problem, this could be the best we can do, I need to think more about it, but this approach maybe is best in the end.

Request for changes is due to coqdep/coqchk being broken, see CI link.

Indeed, that's a problem of allowing warnings in CI, and having warnings that should be errors :)

Reply:
So @ejgallego coqdep is fixed and apparently coqchk doesn't require any such change since the .vo store full paths.

Reply:
@ejgallego I believe this is your request for changes

Reply:
I see, thanks! I had also one related to not losing build caching, so I didn't remember this one.

I'd suggest to first merge #16079 tho, as otherwise it is not possible to detect breakage here, IMHO that's very very dangerous. I had scheduled this topic for the last call, but it was overtaken by other topics.

I'd recommend addressing it in the call today so we can move forward.